utbot-framework-api/src/main/kotlin/org/utbot/framework/plugin/api/Api.kt

```diff
@@ -140,7 +140,27 @@ abstract class UtExecution(
     var summary: List<DocStatement>? = null,
     var testMethodName: String? = null,
     var displayName: String? = null
-) : UtResult()
+) : UtResult() {
+    abstract fun copy(
+        stateBefore: EnvironmentModels = this.stateBefore,
+        stateAfter: EnvironmentModels = this.stateAfter,
+        result: UtExecutionResult = this.result,
+        coverage: Coverage? = this.coverage,
+        summary: List<DocStatement>? = this.summary,
+        testMethodName: String? = this.testMethodName,
+        displayName: String? = this.displayName,
+        // [UtSymbolicExecution]
+        instrumentation: List<UtInstrumentation>? = null,
+        path: MutableList<Step>? = null,
+        fullPath: List<Step>? = null,
+        // [PythonUtExecution]
+        stateInit: EnvironmentModels? = null,
+        diffIds: List<Long>? = null,
+        // [UtFuzzedExecution]
+        fuzzingValues: List<Any>? = null, // FuzzedValue
+        fuzzedMethodDescription: Any? = null, // FuzzedMethodDescription
+    ): UtExecution
```

Comment:
I don't think it's a good design decision, to make `UtExecution` aware of all these properties from its subclasses. If needed, subclasses can add overloads of `copy` method with more parameters. 
```suggestion
    abstract fun copy(
        stateBefore: EnvironmentModels = this.stateBefore,
        stateAfter: EnvironmentModels = this.stateAfter,
        result: UtExecutionResult = this.result,
        coverage: Coverage? = this.coverage,
        summary: List<DocStatement>? = this.summary,
        testMethodName: String? = this.testMethodName,
        displayName: String? = this.displayName,
    ): UtExecution
```

Reply:
If you are referring to this segment of code, then it won't require using `when` since you are only changing `coverage` which base `UtExecution` is aware of 
```kotlin
execution.copy(
    coverage = Coverage(
        coveredInstructions = filteredCoveredInstructions,
        instructionsCount = coverage.instructionsCount,
        missedInstructions = filteredMissedInstructions
    )
)
```

Reply:
To be more clear here's what I suggest:
```kotlin
abstract class UtExecution(
    val stateBefore: EnvironmentModels,
    val stateAfter: EnvironmentModels,
    val result: UtExecutionResult,
    val coverage: Coverage? = null,
    var summary: List<DocStatement>? = null,
    var testMethodName: String? = null,
    var displayName: String? = null
) : UtResult() {
    abstract fun copy(
        stateBefore: EnvironmentModels = this.stateBefore,
        stateAfter: EnvironmentModels = this.stateAfter,
        result: UtExecutionResult = this.result,
        coverage: Coverage? = this.coverage,
        summary: List<DocStatement>? = this.summary,
        testMethodName: String? = this.testMethodName,
        displayName: String? = this.displayName,
    ): UtExecution
}

class UtSymbolicExecution(
    stateBefore: EnvironmentModels,
    stateAfter: EnvironmentModels,
    result: UtExecutionResult,
    val instrumentation: List<UtInstrumentation>,
    val path: MutableList<Step>,
    val fullPath: List<Step>,
    coverage: Coverage? = null,
    summary: List<DocStatement>? = null,
    testMethodName: String? = null,
    displayName: String? = null,
    /** Convenient view of the full symbolic path */ val symbolicSteps: List<SymbolicStep> = listOf(),
) : UtExecution(stateBefore, stateAfter, result, coverage, summary, testMethodName, displayName) {
    override fun copy(
        stateBefore: EnvironmentModels,
        stateAfter: EnvironmentModels,
        result: UtExecutionResult,
        coverage: Coverage?,
        summary: List<DocStatement>?,
        testMethodName: String?,
        displayName: String?
    ): UtExecution = UtSymbolicExecution(
        stateBefore = stateBefore,
        stateAfter = stateAfter,
        result = result,
        instrumentation = instrumentation,
        path = path,
        fullPath = fullPath,
        coverage = coverage,
        summary = summary,
        testMethodName = testMethodName,
        displayName = displayName
    )

    fun copy(
        stateBefore: EnvironmentModels = this.stateBefore,
        stateAfter: EnvironmentModels = this.stateAfter,
        result: UtExecutionResult = this.result,
        coverage: Coverage? = this.coverage,
        summary: List<DocStatement>? = this.summary,
        testMethodName: String? = this.testMethodName,
        displayName: String? = this.displayName,
        instrumentation: List<UtInstrumentation> = this.instrumentation,
        path: MutableList<Step> = this.path,
        fullPath: List<Step> = this.fullPath
    ): UtExecution = UtSymbolicExecution(
        stateBefore = stateBefore,
        stateAfter = stateAfter,
        result = result,
        instrumentation = instrumentation,
        path = path,
        fullPath = fullPath,
        coverage = coverage,
        summary = summary,
        testMethodName = testMethodName,
        displayName = displayName
    )
}

// more subclasses
```